File: api/src/main/java/com/idoceb00/elogap/domain/Activity.java
package com.idoceb00.elogap.domain;

import java.time.Instant;

public record Activity(
    String id,
    String playerId,
    MatchResult result,
    String queueType,
    String champion,
    String role,
    int kills,
    int deaths,
    int assists,
    int cs,
    int duration, // seconds
    int damage,
    int vision,
    Instant playedAt

) {}

File: api/internal/domain/activity.go
package domain

import "time"

type MatchResult string

const (
	ResultWin  MatchResult = "win"
	ResultLoss MatchResult = "loss"
)

type Activity struct {
	ID        string      `json:"id"`
	PlayerID  string      `json:"playerId"`
	Result    MatchResult `json:"result"`
	QueueType string      `json:"queueType"`
	Champion  string      `json:"champion"`
	Role      string      `json:"role"`
	Kills     int         `json:"kills"`
	Deaths    int         `json:"deaths"`
	Assists   int         `json:"assists"`
	CS        int         `json:"cs"`
	Duration  int         `json:"duration"` // seconds
	Damage    int         `json:"damage"`
	Vision    int         `json:"vision"`
	PlayedAt  time.Time   `json:"playedAt"`
}
